workflow/scripts/realisation_to_srf.py

```diff
@@ -552,23 +676,34 @@ def generate_srf(
     )
 
     environment = SRFEnvironmentContext(
-        work_directory=work_directory, genslip_path=genslip_path, seeds=seeds
+        work_directory=work_directory,
+        genslip_path=genslip_path,
+        generic_slip2srf_path=generic_slip2srf_path,
+        seeds=seeds,
     )
 
-    generate_fault_srfs_multi(
-        source_config.source_geometries,
-        params,
-        environment,
-        single_threaded=single_threaded,
-    )
     srf_name = normalise_name(metadata.name)
-    stitch_srf_files(
-        source_config.source_geometries,
-        rupture_propagation,
-        work_directory,
-        srf_name,
-    )
-    srf_config.write_to_realisation(realisation_ffp)
 
-    shutil.copyfile(work_directory / (srf_name + ".srf"), output_srf_filepath)
-    realisations.append_log_entry(realisation_ffp)
+    for name, geometry in params.source_config.source_geometries.items():
+        if isinstance(geometry, Fault):
+            generate_fault_srfs_multi(
+                source_config.source_geometries,
+                params,
+                environment,
+                single_threaded=single_threaded,
+            )
+
+            stitch_srf_files(
+                source_config.source_geometries,
+                rupture_propagation,
+                work_directory,
+                srf_name,
+            )
+
+            shutil.copyfile(work_directory / (srf_name + ".srf"), output_srf_filepath)
+
+        else:
+            generate_point_source_srf(name, params, environment, output_srf_filepath)
+
+        realisations.append_log_entry(realisation_ffp)
+        srf_config.write_to_realisation(realisation_ffp)
```

Comment:
This code is still broken! It should remain unchanged from the original.
```suggestion
    generate_fault_srfs_multi(
        source_config.source_geometries,
        params,
        environment,
        single_threaded=single_threaded,
    )
    srf_name = normalise_name(metadata.name)
    stitch_srf_files(
        source_config.source_geometries,
        rupture_propagation,
        work_directory,
        srf_name,
    )
    srf_config.write_to_realisation(realisation_ffp)

    shutil.copyfile(work_directory / (srf_name + ".srf"), output_srf_filepath)
    realisations.append_log_entry(realisation_ffp)
```

Instead `generate_fault_srfs_multi` needs to be modified to generate point source SRFs.